kedro/framework/cli/micropkg.py

```diff
@@ -299,13 +299,27 @@ def _get_fsspec_filesystem(location: str, fs_args: Optional[str]):
         return None
 
 
+def _is_within_directory(directory, target):
+    abs_directory = directory.resolve()
+    abs_target = target.resolve()
+    return abs_directory in abs_target.parents
+
+
+def safe_extract(tar, path):
+    for member in tar.getmembers():
+        member_path = path / member.name
+        if not _is_within_directory(path, member_path):
+            raise Exception("Attempted Path Traversal in Tar File")
```

Comment:
I know @TrellixVulnTeam added this exception message, but I think we can make it a bit clearer. Maybe something like "Failed to safely extract tar file"